### ISSUE TITLE
Difference in metrics between the two arxiv version

### ISSUE THREAD

**Issue by RoyHirsch** [CAMMA-public/SelfSupSurg#1]
Hi, I've noticed there is a difference in the F1 score for phase recognition between the two arxiv versions and wanted to understand the differences. It seems that the linear evaluation protocol haven't changed so how can you explain the differences in tables 3 and 4?

For example for table 3: dino, base, 40 videos -> 71.6 and in the older version the value was 67.0.

Thanks for your great work!

**Comment by DpkApt** [comment]
Hello @RoyHirsch, this is due to an error in our initial metric computation consistently affecting both baseline methods and our proposed methods for phase recognition. We have made corrections to both the paper and repository to reflect this. The evaluation code is now also public to facilitate reproducibility. Please reference only the updated arXiv results. We sincerely apologize for any inconvenience.

Also for your reference, unlike the hyperparameter study, the tables you mentioned (3&4) do not use a linear evaluation protocol, as described in Section 3.3. Instead, the entire network is fine-tuned (see Section 3.4).

**Comment by RoyHirsch** [comment]
@DpkApt thank you for the detailed answer. 

1) I guess the fine-tuning of the TCN is not in conjunction with the resnet weights. In that case, you first fine-tune the resnet and then use its representation to train a TCN?
2) As I understand, the function for calculating the phase recognition F1 is `utils.accuracy` which implements a micro F1 averaging (gives equal importance to each observation, in compare macro F1 gives each phase) right? 
3) For further clarification, does the results in table 3&4 are from the validation or the train set?

**Comment by DpkApt** [comment]
No problem @RoyHirsch. Attaching some comments below corresponding to your questions

1. You are right, for the TCN the training is in two stages following the work "TeCNO: Surgical phase recognition with multi-stage temporal convolutional networks"
2. I'm not sure what utils.accuracy refers to, but the function we use is available in utils.generate_test_results.py. Basically, we calculate F1 for each video and phase, average by phase, and then across videos.  (Note: the ordering matters if we don't have a universal representation of all phases in all procedures)
3. Tables 3 & 4 are test set results. Only Table 2 shows validation set results (no model selection in this case though).

**Comment by RoyHirsch** [comment]
Thank you @DpkApt  for the detailed answer!
